requre/helpers/guess_object.py

```diff
@@ -0,0 +1,127 @@
+# MIT License
+#
+# Copyright (c) 2018-2019 Red Hat, Inc.
+
+# Permission is hereby granted, free of charge, to any person obtaining a copy
+# of this software and associated documentation files (the "Software"), to deal
+# in the Software without restriction, including without limitation the rights
+# to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+# copies of the Software, and to permit persons to whom the Software is
+# furnished to do so, subject to the following conditions:
+#
+# The above copyright notice and this permission notice shall be included in all
+# copies or substantial portions of the Software.
+#
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+# IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+# FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+# AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+# LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+# OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+# SOFTWARE.
+
+import yaml
+import logging
+import pickle
+
+from typing import Any, Dict, Callable
+from requre.objects import ObjectStorage, original_time
+from requre.helpers.simple_object import Simple, Void
+
+from requre.cassette import Cassette, CassetteExecution, StorageKeysInspectFull
+
+logger = logging.getLogger(__name__)
+GUESS_STR = "guess_type"
+
+
+class Guess(ObjectStorage):
+    @classmethod
+    def guess_type(cls, value):
+        try:
+            # Try to use type for storing simple output (list, dict, str, nums, etc...)
+            with open("/dev/null", "w") as output_null:
+                yaml.safe_dump(value, output_null)
+            return Simple
+        except Exception:
+            try:
+                # Try to store anything serializable via pickle module
+                pickle.dumps(value)
+                return ObjectStorage
+            except Exception:
+                # do not store anything if not possible directly
+                # TODO: this part is risky, because it can cause bad replaying
+                # TODO: in case output is expected and checked
+                return Void
```

Comment:
What about some error in that case?

Reply:
Question is if this is bad. Both solution, with/without exception have pros and cons, maybe just add there `warning.warn` to output, to have it mentioned, that this situation happened.